logic/captcha_answer.go

```diff
@@ -0,0 +1,142 @@
+package logic
+
+import (
+	"encoding/json"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/allegro/bigcache/v3"
+	tb "gopkg.in/tucnak/telebot.v2"
+)
+
+// This is the handler for listening to incoming user message.
+// It will uh... do a pretty long task of validating the input message.
+func (d *Dependencies) WaitForAnswer(m *tb.Message) {
+	// Check if the message author is in the captcha:users list or not
+	// If not, return
+	// If yes, check if the answer is correct or not
+	check, err := userExists(d.Cache, strconv.Itoa(m.Sender.ID))
+	if err != nil {
+		handleError(err, d.Logger, d.Bot, m)
+		return
+	}
+
+	if !check {
+		return
+	}
```

Comment:
```suggestion
	exists, err := userExists(d.Cache, strconv.Itoa(m.Sender.ID))
	if err != nil {
		handleError(err, d.Logger, d.Bot, m)
		return
	}

	if !exists {
		return
	}
```suggestion
	exists, err := userExists(d.Cache, strconv.Itoa(m.Sender.ID))
	if err != nil {
		handleError(err, d.Logger, d.Bot, m)
		return
	}

	if !exists {
		return
	}
```